ocp_resources/benchmark.py

```diff
@@ -0,0 +1,95 @@
+import logging
+
+from openshift.dynamic.exceptions import NotFoundError
+
+from ocp_resources.resource import NamespacedResource
+from ocp_resources.utils import TimeoutSampler
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Benchmark(NamespacedResource):
+    """
+    Benchmark resource
+    Defined by https://github.com/cloud-bulldozer/benchmark-operator
+
+    The benchmark-operator monitors the `my-ripsaw` namespace for `Benchmark` resources.
+    When a new `Benchmark` is created, the benchmark-operator creates and starts the pods or VMs necessary,
+    and triggers the benchmark run.
+    """
+
+    api_group = NamespacedResource.ApiGroup.RIPSAW_CLOUDBULLDOZER_IO
+
+    class Status:
+        NONE = None  # None state is valid for newly created benchmark resources
+
+    class Workload:
+        class Kind:
+            VM = "vm"
+            POD = "pod"
+
+    def _wait_for_instance_key(self, parent, key):
+        """
+        Wait for key to exist in parent attribute of instance
+
+        Args:
+            parent (str): An attribute of self.instance that should contain key
+            key (str): A dictionary entry within parent
+
+        Returns:
+            Value of key if found, otherwise None
```

Comment:
Missing type in return docstring
str or None: Value of key if found, otherwise None